sdk/azidentity/client_secret_credential_test.go

```diff
@@ -42,7 +43,11 @@ func TestClientSecretCredential_GetTokenSuccess(t *testing.T) {
 func TestClientSecretCredential_Live(t *testing.T) {
 	opts, stop := initRecording(t)
 	defer stop()
-	o := ClientSecretCredentialOptions{ClientOptions: opts}
+	disableID, err := strconv.ParseBool(disableInstanceDiscovery)
```

Comment:
Could potentially move this logic into livetest, but I elected to put it here due to error handling if the env variable is empty (aka most cases), and it would just set disableID to be false (default value)